multi-tenancy-observability/README.md

```diff
@@ -0,0 +1,270 @@
+
+##### Namespace == Tenant
+
+This approach makes it really easy for us and customers to configure tenants and it would definitely be easy to onboard teams (e.g. adding a label on the namespace would make the agents collect logs from the containers in the namespace).
+
+But this approach brings a few questions:
+1. What tenancy should common internal apps use (e.g. prometheus-operator, api-server, cilium logs are accessed by multiple internal teams)? Do we need to ensure customers are putting those apps into a namespace accessible by all?
+2. What about teams that share namespaces?
+
+##### App == Tenant
+
+This approach is especially flexible for customers but we do not think this approach makes sense because the mapping configuration would be impossible to maintain and we would probably face technical limits pretty fast
+
+##### Hybrid mode
+
+This approach would allow customers to be able to set the tenancy as they want (per namespace or per app depending on the use case).
+
+#### Implementation details
+
+The implementation details are at this stage are still an open topic, because no single tool fulfills all of our requirements. Until this RFC we use two tools, grafana-agent and promtail, which are bot not yet feature complete.
+One preferred alternative would be [PodLogs](https://grafana.com/docs/agent/latest/operator/api/#podlogs-a-namemonitoringgrafanacomv1alpha1podlogsa), the equivalent of a Service or PodMonitor for metrics, but the existing implementation at the time of this RFC is not yet fully compliant with our needs as well.
+
+We envisionned different distinct approaches here:
+
+#### Approach 1: Customers provide their logs themselves
+
+In this approach, based on the tenant information that the customer provided to us, we would generate a user/password pair that the customer would then use in their favorite log shipper to send their logs to Loki.
+
+Pros:
+- This is quite easy for us to implement as we already have the password generation logic
+- We allow customers to use the tool they know and love.
+- We will not receive pages if the tool is not working properly
+
+Cons:
+- This is far from the idea of a platform
+- There is a high risk that configuration will be messy and hard for use to debug when we need to
+
+This approach might be useful for customers that want to send logs from outside the clusters
+
+#### Approach 2: Extract tenant information from a label/PodLog
+
+Existing logging agents support setting the tenant to send to loki using a custom log label.
+We can configure our logging agents to get the logs of specific namespaces/pods if they have a label like `giantswarm.io/tenant: my-tenant` or using a PodLog CR (if applicable).
+
+Pros:
+- Quite easy to configure for logs
+
+Cons:
+- Tenant configuration will need to be configured in multiple places (MC for the read path and WC for the write path)
```

Comment:
Might not be a big problem as read path is access management and MC is a good place for that and write path workload related so deploying together with workload makes sense. The two lifecycles are separated.